cli/command/service/opts.go

```diff
@@ -660,6 +667,109 @@ func (options *serviceOptions) makeEnv() ([]string, error) {
 	return currentEnv, nil
 }
 
+func (options *serviceOptions) ToPrivileges(flags *pflag.FlagSet) (*swarm.Privileges, error) {
+	// we're going to go through several possible uses of the Privileges
+	// struct, which may or may not be used. If some stage uses it (after the
+	// first), we'll check if it's nil and create it if it hasn't been created
+	// yet.
+	var privileges *swarm.Privileges
```

Comment:
Nit: when the function returns no error, is there any meaningful distinction between `nil` Privileges and a pointer to a zero-valued struct? If so, please document. Otherwise, this function's implementation could be simplified by pre-initializing the struct unconditionally.
```suggestion
	privileges := &swarm.Privileges{}
```